### ISSUE TITLE
Improve group selection for entries

### ISSUE THREAD

**Issue by hpoul** [authpass/authpass#75]

**Comment by cvila84** [comment]
Do you confirm it means you plan to support displaying the entries as a tree ? Sometimes, I look up for my passwords in browsing the corresponding groups instead of searching. Thanks !

**Comment by hpoul** [comment]
@cvila84 nope, this issue is actually for moving entries to a different group, not for displaying.. but i think it should be closed already, it already uses a new selection UI.

as for displaying entries as a tree: this is  currently not planned. Is it really that much more work to click on the filter icon if you really need to browse  by group?
Sorry for my ignorance, but to be honest i am not able to imagine a use case for the "browsing by group", maybe you can give more details. The way I work with passwords is: 1. I visit a website where i've got an account 2. look up that account in my passwords by typing the URL/service into the search 3. using the username/password ... I can't figure out a common use case why I would browse through my passwords without searching for a specific entry?

**Comment by cvila84** [comment]
thanks to your message, I just discovered we can filter by group, which helps a bit for my use-case.

this use-case is : I go on my online bank with their visual numeric keyboard to enter my login, I may not know their URL and I don't care since usually I reach the site with my bookmarks, so here I'm opening my DB and select my bank group in the tree and tada I can select in the list the account I want to login to and I just read the login and password info.

i fully understand your current approach based on search, which are the standard in nowadays tools but sometimes, as for the above use-case, searching is not the quickest way (in term of UX "number of clicks")

**Comment by hpoul** [comment]
well, you don't really need to search for URL, just search for the name of the bank..
btw, the search field also includes the name of the groups, so you could even just search for 'bank' to get all the entries in your 'bank' group :-)

imho adding another core password list UI based on a tree structure would just add more clutter.. I think the current approach of having a single list of passwords and another one to filter by groups is superior, if you are not coming at it with the expectation that everything works like in (some) other keepass compatible apps.
maybe there is some way to improve the discoverability of the filter? It seems like you found this issue before finding the filter UI :-)

**Comment by cvila84** [comment]
"if you are not coming at it with the expectation that everything works like in (some) other keepass compatible apps". I think the key point is here. I came with my habits inherited from keepass (and minikeepass on IOS). I'm OK to change my habits because I understand your approach, but you have to be sure that the 1000+ users that now MUST leave Minikeepass on IOS to come to your app will do the same :)

**Comment by hpoul** [comment]
Maybe I should just write up a "AuthPass for Minikeepass Users" help page 😅️
Since the kdbx file contains a `Generator` field with "MiniKeePass" I could even show a help banner or something :-D

**Comment by hpoul** [comment]
have you experienced other hurdles for migrating away from minikeepass?

**Comment by hpoul** [comment]
I have been experimenting a bit, and trying to put group selection into an app drawer.. i think it should be certainly better than hiding it behind the filter -> customize app bar menu:

![TameFoolhardyFrilledlizard-size_restricted](https://user-images.githubusercontent.com/313066/86588363-dc8fea80-bf8b-11ea-983c-9ee6f0f72c58.gif)

**Comment by Skymirrh** [comment]
> imho adding another core password list UI based on a tree structure would just add more clutter.. I think the current approach of having a single list of passwords and another one to filter by groups is superior, if you are not coming at it with the expectation that everything works like in (some) other keepass compatible apps.

Hello! I understand your point, however in my experience this is not an issue of expectations, but of how people think. I got a bit carried away writing this so apologies if it's too long/irrelevant.

In a previous job I often had to design and implement "central item lists", i.e. main screens where users would spend more than 50% of application usage time, akin to AuthPass' password list, for quickly searching/accessing huge amount of data (construction defects, etc.).

The "flat structure" as found in AuthPass, i.e. a list where by default everything is displayed and the user is then able to filter/search, was one of the main variants I would end up with after prototyping variants with end users, as it is very powerful when dealing with huge lists. However, I found that less tech-savvy users can very easily be confused by it, as they would be very efficient when using "Amazon-like" filters (e.g. UI with multiple properties/tags, which combine using `AND` filters, and where you can tick zero, one, or more values for each property/tag, which combine using `OR` filters) but not for what people thought of as types / groups / categories.

As an example, in prototypes they would almost always efficiently filter some specific type of `foo` items with `(propA = something OR propA = somethingElse) AND (amount > 300)`. However that assumes that they were already browsing `foo` items. For some reason, even though it is strictly the same thing, when type was made a regular property of a flat list they would routinely not immediately think of doing `(type = foo) AND (propA = something OR propA = somethingElse) AND (amount > 300)` and require help to get them started on how to filter. If given the choice of both a flat structure that can be filtered by type, and a tree structure based on types, after some time they would almost never use the flat structure, and always the tree structure.

Anecdotally, this is consistent with my own experience introducing KeePass to my family over the last year or so. I host KeePass databases for my family, with which they can interact on all their devices (PC, phones, tablets) via WebDAV synchronization. In my small sample (4 people), where none of them had prior KeePass exposure, they all gave me the same feedback regarding groups in AuthPass: "I don't understand how it works" and after explanation "I have trouble showing only entries from a specific group". At the same time, they had absolutely no trouble using groups on KeePass desktop, which uses a tree structure. Since this is a major hurdle for them, I had no choice but to recommend alternatives with tree structures for now (Strongbox on iOS, Keepass2Android on Android).

While the latest iteration via the app drawer above is definitely an improvement on the previous hidden UI, I think it would still make more sense for the average user to have a tree structure similar to KeePass' structure. Perhaps we can have both via a toggle allowing either multi-select (checkboxes) and single-select (no checkboxes, only highlight of currently selected)?
